lib/ssl/src/tls_connection.erl

```diff
@@ -580,24 +580,28 @@ init({call, From}, {start, Timeout},
             handshake_env = #handshake_env{renegotiation = {Renegotiation, _}} = HsEnv,
             connection_env = CEnv,
 	    ssl_options = #{log_level := LogLevel,
-                            %% Use highest version in initial ClientHello.
-                            %% Versions is a descending list of supported versions.
-                            versions := [HelloVersion|_] = Versions,
-                            session_tickets := SessionTickets} = SslOpts,
-	    session = NewSession,
-	    connection_states = ConnectionStates0
-	   } = State0) ->
+                        %% Use highest version in initial ClientHello.
+                        %% Versions is a descending list of supported versions.
+                        versions := [HelloVersion|_] = Versions,
+                        session_tickets := SessionTickets,
+                        ocsp_responder_certs := OcspResponderCerts,
```

Comment:
In this function we should first match out the option `ocsp_stapling` to see if stapling is enabled. If it is enabled and only if it is enabled we calculate the OcspNonce. `public_key:ocsp_nonce` could be wrapped in another function that takes care of this. Current code will always calculate it even if `ocsp_stapling` is set to false.

Reply:
> In this function we should first match out the option `ocsp_stapling` to see if stapling is enabled. If it is enabled and only if it is enabled we calculate the OcspNonce. `public_key:ocsp_nonce` could be wrapped in another function that takes care of this. Current code will always calculate it even if `ocsp_stapling` is set to false.

done